crates/astria-celestia-client/src/blob_space.rs

```diff
@@ -35,12 +35,195 @@ pub fn celestia_namespace_v0_from_hashed_bytes(bytes: &[u8]) -> Namespace {
+            // variants Sha256([u8; 32]) or None.
+            return Err(SequencerNamespaceDataConstruction::MissingDataHash);
+        };
+
+        let calculated_block_hash = header.hash();
+        if block_hash != calculated_block_hash {
+            return Err(SequencerNamespaceDataConstruction::HashOfHeaderBlockHashMismatach);
+        }
+
+        let action_tree_root_hash = sha2::Sha256::digest(action_tree_root);
+        if !action_tree_root_inclusion_proof.verify(&action_tree_root_hash, data_hash) {
+            return Err(SequencerNamespaceDataConstruction::ActionTreeRootVerification);
+        }
+
+        sequencer_types::sequencer_block_data::assert_chain_ids_are_included(
+            chain_ids_commitment,
+            &chain_ids_commitment_inclusion_proof,
+            &rollup_chain_ids,
+            data_hash,
+        )
+        .map_err(Into::<SequencerNamespaceDataConstruction>::into)?;
+
+        Ok(Self {
+            block_hash,
+            header,
+            rollup_chain_ids,
+            action_tree_root,
+            action_tree_root_inclusion_proof,
+            chain_ids_commitment,
+            chain_ids_commitment_inclusion_proof,
+        })
+    }
+
+    /// Return the data hash of the cometbft header stored in the blob.
+    ///
+    /// # Panics
+    /// This method panics if a variant of the [`SequencerNamespaceData`] was violated.
+    #[must_use]
+    pub fn data_hash(&self) -> [u8; 32] {
+        let Some(Hash::Sha256(data_root)) = self.header.data_hash else {
+            panic!(
+                "data_hash must be set; this panicking means an invariant of \
+                 SequencerNamespaceData was violated. This is a bug"
+            );
+        };
+        data_root
+    }
+
+    /// Returns the cometbft block hash stored in the blob.
+    #[must_use]
+    pub fn block_hash(&self) -> Hash {
+        self.block_hash
+    }
+
+    /// Returns the cometbft header stored in the blob.
+    #[must_use]
+    pub fn header(&self) -> &Header {
+        &self.header
+    }
+
+    /// Returns the list of rollup chain IDs stored in the blob.
+    #[must_use]
+    pub fn rollup_chain_ids(&self) -> &[ChainId] {
+        &self.rollup_chain_ids
+    }
+
+    /// Returns the Merkle tree hash of the tree constructed from the transactions
+    /// stored in the blob.
+    #[must_use]
+    pub fn action_tree_root(&self) -> [u8; 32] {
+        self.action_tree_root
+    }
+
+    /// Returns the proof that the action tree root was included in the cometbft sequencer block.
+    ///
+    /// [`Self::data_hash`] is the Merkle tree hash this proof (together with the sha256 hash
+    /// of [`Self::action_tree_root`]) is evaluated against.
+    #[must_use]
+    pub fn action_tree_root_inclusion_proof(&self) -> &merkle::Proof {
+        &self.action_tree_root_inclusion_proof
+    }
+
+    /// Returns the Merkle tree hash construced from [`Self::rollup_chain_ids`].
```

Comment:
```suggestion
    /// Returns the Merkle tree hash constructed from [`Self::rollup_chain_ids`].
```

Reply:
The term `Merkle tree hash` and `Merkle tree root` can indeed used synonymously. [RFC 6962](https://datatracker.ietf.org/doc/html/rfc6962) defines the Merkle Tree Hash recursively in terms of the Merkle Tree Hashes of its subtrees and never refers to "Merkle Tree Root Hash". Indeed, it does an explanation toward the bottom of a struct field:
>    "sha256_root_hash" is the root of the Merkle Hash Tree.

But to be consistent with their naming I am going to stick with `Merkle Tree Hash` (and will capitalize it here).